### PR TITLE
DOC: Support for readthedocs

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #37

- Adds basic support for readthedocs based on `pysatNASA` implementation, including tests.
- Cleans up docstring formatting

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Building docs locally and inspecting this branch at https://pysatcdaac.readthedocs.io/en/doc-37_rtd/

**Test Configuration**:
* Operating system: Monterrey
* Version number: Python 3.10.9
* Sphinx 6.2.1

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

### CODE REVIEW

**Comment by JonathonMSmith** [comment]
The example code produces a couple of warnings when I run it. I assume this is okay though.
I get the "use_header" warning and the "No cleaning routines" warning.

**Comment by jklenzing** [comment]
Those are expected. We could use `use_header=True` for the example to clean this up a little.

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = python -msphinx
```
Comment (aburrell) [inline docs/Makefile:6]:
```suggestion
SPHINXBUILD   = python -m sphinx
```
For clarity of flag and flag input.

File: docs/overview.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+========
+
+This is a library of ``pysat`` instrument modules and methods designed to support
+NASA instruments and missions archived at the Community Data Analysis Web
+portal.
+
+.. image:: figures/logo.png
+    :width: 400px
+    :align: center
+    :alt: pysatNASA Logo, a blue planet with multiple orbiting pythons and the module name superimposed
```
Comment (aburrell) [inline docs/overview.rst:11]:
```suggestion
    :alt: pysatCDDAC Logo, a blue planet with multiple orbiting pythons and the module name superimposed
```

File: docs/overview.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Overview
+========
+
+This is a library of ``pysat`` instrument modules and methods designed to support
+NASA instruments and missions archived at the Community Data Analysis Web
```
Comment (aburrell) [inline docs/overview.rst:5]: Needs to be correct.

File: docs/supported_instruments.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+COSMIC GPS
+----------
+
+.. automodule:: pysatCDAAC.instruments.cosmic_gps
+   :members:
+
+.. _cosmic2_ivm:
+
+COSMIC2 IVM
```
Comment (aburrell) [inline docs/supported_instruments.rst:14]:
```suggestion
COSMIC-2 IVM
```

File: docs/overview.rst (hunk 3 above)
Comment (jklenzing) [reply docs/overview.rst:5]: caught in the index, but not the overview. 🤦

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
+.. pysatCDAAC documentation main file.  Remember that it should at least
+   contain the root `toctree` directive.
+
+Welcome to the pysatCDAAC documentation
+=======================================
+
+This documentation describes the pysatCDAAC module, which contains
+routines to space science data hosted at the COSMIC Data Analysis and Archive
+Center (CDAAC)as pysat.Instrument objects.
```
Comment (jklenzing) [inline docs/index.rst:9]:
```suggestion
Center (CDAAC) as pysat.Instrument objects.
```

File: docs/examples/ex_init.rst [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+   import datetime as dt
+   import pysat
+   import pysatCDAAC as py_cdaac
+
+   pysat.utils.registry.register_by_module(py_cdaac.instruments)
+
+   date = dt.datetime(2021, 1, 1)
+   ivm = pysat.Instrument(platform='cosmic2', name='ivm',
+                          inst_id='e1', update_files=True)
```
Comment (jklenzing) [inline docs/examples/ex_init.rst:21]:
```suggestion
   ivm = pysat.Instrument(platform='cosmic2', name='ivm',
                          inst_id='e1', update_files=True,
                          use_header=True)
```

**Comment by jklenzing** [comment]
> Those are expected. We could use `use_header=True` for the example to clean this up a little.

Updated

File: docs/examples/ex_init.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
+Loading COSMIC2 IVM data
+========================
+
+pysatCDAAC uses `pysat <https://github.com/pysat/pysat>`_ to load
+space science instrument data.  As specified in the
+`pysat tutorial <https://pysat.readthedocs.io/en/latest/tutorial.html>`_,
+data may be loaded using the following commands.  Data from the Ion Velocity
+Meter on board the Ionospheric CONnection Explorer `(ICON) <https://www.nasa.gov/icon>`_ is used as an example.
```
Comment (aburrell) [inline docs/examples/ex_init.rst:8]: Line too long

Comment (jklenzing) [reply docs/examples/ex_init.rst:8]: also inaccurate. 🙀

**Review by aburrell — APPROVED** [review]

**Comment by jklenzing** [comment]
Thanks @JonathonMSmith and @aburrell for the reviews!

**Pull request merged**